lib/g_searcher/search_results/report.ex

```diff
@@ -0,0 +1,29 @@
+defmodule GSearcher.SearchResults.Report do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias GSearcher.Accounts.User
+  alias GSearcher.SearchResults.{ReportSearchResult, SearchResult}
+
+  schema "reports" do
+    field :title, :string
+
+    field :csv_path, :string, virtual: true
+
+    belongs_to :user, User
+
+    has_many :report_search_results, ReportSearchResult
+    many_to_many :search_results, SearchResult, join_through: "reports_search_results"
+
+    timestamps()
+  end
+
+  def create_changeset(report \\ %__MODULE__{}, attrs) do
+    report
+    |> cast(attrs, [:title, :csv_path, :user_id])
+    |> validate_required([:title, :user_id])
+    |> unique_constraint([:title, :user_id],
```

Comment:
Although this PR was merged but I think this is an unnecessary feature as a user could update the same `file` to get the result, think about the user would like to check the result for a list of keyword day by day, so they could upload the same file hehe, with this constraint, it's kind of blocking the user function, I think we could remove this constraint validation, what do you think @bterone.

We could display something like

```
Report id, title, date
1, campaign_1, 05-01-2021
2, campaign_1, 06-01-2021 
```

What do you think :)

Reply:
Note, by title, it does not mean the `CSV file_name`, it's the `user input` for the report title.

So they can keep uploading the same file if they want 😂 
it's only for on the `report index page` where I think it might be more confusing for the user if they are given the ability to write the same `report title` and then they could see two rows with the same title.

Reply:
Ah I see :)